stylix/palette.nix

```diff
@@ -91,11 +156,79 @@ in
+                v3 = builtins.fromJSON (builtins.elemAt values 4);
+                preLightness = (v1 + v2 + v3) / 3.0;
+                adj =
+                  (preLightness / 255.0 * (1.0 - primaryScale) + primaryScale) / preLightness
+                  * 255.0;
+                v1adj = lib.max (lib.min (v1 * adj) 255.0) 0.0;
+                v2adj = lib.max (lib.min (v2 * adj) 255.0) 0.0;
+                v3adj = lib.max (lib.min (v3 * adj) 255.0) 0.0;
+                round =
+                  x:
+                  let
+                    floored = builtins.floor x;
+                    diff = x - floored;
+                  in
+                  if diff >= 0.5 then floored + 1 else floored;
+              in
+              toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v1adj)))
+              + toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v2adj)))
+              + toString (lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v3adj)));
+            jsonData = lib.importJSON cfg.generated.json;
+            colors =
+              if cfg.themeGeneration.polarity == "light" then
+                jsonData.colors.light
+              else
+                jsonData.colors.dark;
+          in
+          if cfg.themeGeneration.polarity == "light" then
+            {
+              base00 = adjustLightness colors.background cfg.themeGeneration.primaryScale.dark;
+              base01 = adjustLightness colors.surface_container cfg.themeGeneration.primaryScale.dark;
+              base02 = adjustLightness colors.surface_container_highest cfg.themeGeneration.primaryScale.dark;
+              base03 = adjustLightness colors.outline cfg.themeGeneration.primaryScale.dark;
+              base04 = adjustLightness colors.on_surface_variant cfg.themeGeneration.primaryScale.dark;
+              base05 = adjustLightness colors.on_surface cfg.themeGeneration.primaryScale.dark;
+              base06 = adjustLightness colors.on_secondary_fixed cfg.themeGeneration.primaryScale.dark;
+              base07 = adjustLightness colors.on_primary_container cfg.themeGeneration.primaryScale.dark;
+              base08 = adjustLightness colors.error cfg.themeGeneration.primaryScale.dark;
+              base09 = adjustLightness colors.on_tertiary cfg.themeGeneration.primaryScale.dark;
+              base0A = adjustLightness colors.on_secondary_container cfg.themeGeneration.primaryScale.dark;
+              base0B = adjustLightness colors.on_secondary_fixed_variant cfg.themeGeneration.primaryScale.dark;
+              base0C = adjustLightness colors.on_primary_fixed cfg.themeGeneration.primaryScale.dark;
+              base0D = adjustLightness colors.surface_tint cfg.themeGeneration.primaryScale.dark;
+              base0E = adjustLightness colors.on_tertiary_fixed cfg.themeGeneration.primaryScale.dark;
+              base0F = adjustLightness colors.on_error_container cfg.themeGeneration.primaryScale.dark;
```

Comment:
Is it not supposed to be `cfg.themeGeneration.primaryScale.light`?

Reply:
i changed it to light